docs/PULL_REQUEST_TEMPLATE.md

```diff
@@ -0,0 +1,59 @@
+## Checklist
+<!--
+Mark an `[x]` for completed items.
+-->
+
+* [ ] Update CHANGELOG.rst  for each commit separately.
+* [ ] Bump the process version:
+    * **Last number**: changes that do not affect the api/interface. Examples:
+        bugfix, typo fix, change/add warning messages,..
+    * **Middle number**: bigger changes that don't make it backwards incompatible.
+        Examples: add output field, add non-mandatory input parameter, use
+        a different tool that produces same results...
+    * **First number**: Backwards incompatible changes, are changes that
+        break the api/interface. Examples: renaming the input/output, adding
+        mandatory input, removing input/output…
+* [ ] All inputs are used in process.
+* [ ] All outputs fields have their ``re-save``.
+
+## Additional guidelines
+
+### Processes ###
+
+* Code that was once accepted should be kept like it is. It should be changed
+    only when it's functionality is changed, or small parts may be changed
+    to fix bugs (If you change it only to make it nicer there is a chance
+    to introduce a bug).
+* Set the optimal number of cores that the tool can use (check if parallelization
+    can be performed).
+* Description of tool should be self explanatory with links to tool homepage
+    and publication.
+* Write explanatory input labels and add short commands in [ ] (for example:
+    Maximum length [maxlength]).
+* When possible add default option. If the default is set, ``required: false``
+    is not needed.
+* Add descriptive error and warning messages.
+* After each tool, python or r script it needs to be ``re-checkrc``.
+* Add ``re-progress`` notes.
+
+### Commit and PR messages ###
+
+* Each commit should be minimal (i.e. 1 change) and self-contained (including
+    tests).
+* Changes in docker images needs to be in separate PR.
+* For work in progress add [WIP] at the beginning of commit messages and
+    PR title.
+
+### Writing tests ###
+
+* Add meaningful names of test files with spaces:
+    * Genomes: ``genome species.fa.gz``
+    * Annotations: ``annotation species.gtf.gz``
+    * Adapters: ``adapters-name/type-source``.
+* Output and specific input files should have name of the tested tool.
+* Keep small sizes of test files.
+* When testing workflows, verify workflow success as best as possible.
```

Comment:
Add something like: "Pay attention to edge cases: What will happen is any data is missing, if length of some list is 0, if string is given where one expects integer, if file is empty, ... "